HPt630 tests

philipanda opened the pull request.

philipanda committed:
3mdeb-templates.json: Update products to qubes R4.2.3

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates.json: Update AEM packages

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates.json: Add new producsts & machines

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates: Update jobgroups

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
generalhw: hpt630v1: Add

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
generalhw/hpt630v1: Add +x to scripts

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
generalhw/hpt630v1/power: Fix parsing action to sonoff api

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates: Update MAX_ISO_SIZE

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates: Add "legacy_boot" key to jobs with legacy boot

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
needles: HP GRUB bootmenu needles

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
install_partitioning_default: check for install over existing

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
firstboot & 3mdeb-templates: Workaround for missing video in lukspass

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates.json: Set TIMEOUT_SCALE for hpt630v1 machine

(It's really slow...)

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
tests/aem_hw.pm: Support HPt630v1

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates: Add SKL_VER to aem-setup test suite

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
tests/aem_hw.pm: Running Qubes: update grub traversal

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
aem_hw.pm: Workaround for missing signal on lukspass

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
aem_hw: Set corect boot partition depending on legacy/efi boot

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
aem_hw: Do lukspass no signal workaround in loop to save time

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
aem_hw: Don't do grub2-install on efi systems

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
aem_hw.pm: Change poweroff to reboot on the end of `run`

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
aem_hw: clear_tpm_hp: Add verifying TPM is still enabled after clear

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
aem_hw: Adjust installed packages depending on Legacy/EFI boot

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
needles: Add needles to verify TPM enabled on HP

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates: Fix XEN_VER for installation test suite

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates: Seperate test suites into Legacy/UEFI versions

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
needles: Add needles for aem_hw clearing TPM on HPt630v1

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates: Update START_AFTER for EFI/legacy test variants

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

krystian-hebel commented on generalhw/hpt630v1/README.md:
````diff
@@ -0,0 +1,50 @@
+Perform installation followed by AEM testing:
+
+```
+openqa-cli api -X POST isos DISTRI=qubesos VERSION=4.2 ARCH=x86_64 BUILD=4.2.0 FLAVOR=install-iso-optiplex
````
It would be better to use versions we're testing it with, and also, `optiplex`?

philipanda replied:
Sorry, it was so long since I copy-pasted this README from optiplex that I forgot that it still has to be changed. I'll look through the whole `generalhw/hpt630` again

philipanda replied:
https://github.com/3mdeb/openqa-tests-qubesos/compare/a39198afd3230fdf0e453e356c1fe72dd65ddd95..c1f63647bdb848a222fd9a6d97cbef1d5a4970c9#diff-effe13d6dacca60a03019f6a5c56dba5883bea6674fdbaa74a19f68713474974L14-R14

krystian-hebel commented on generalhw/hpt630v1/README.md:
```diff
@@ -0,0 +1,50 @@
+* `PACKAGES_BASE_URL` - where to look for AEM-related packages.
+* `AEM_VER` - version of `anti-evil-maid` package
+* `GRUB_VER` - version of `grub2-*` packages
+* `XEN_VER` - version of `xen-*` packages
```
Missing `SKL_VER`.

philipanda replied:
https://github.com/3mdeb/openqa-tests-qubesos/compare/a39198afd3230fdf0e453e356c1fe72dd65ddd95..c1f63647bdb848a222fd9a6d97cbef1d5a4970c9#diff-effe13d6dacca60a03019f6a5c56dba5883bea6674fdbaa74a19f68713474974R41

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -50,17 +50,20 @@ use Data::Dumper;
 my $boot_disk;
 my $boot_part;
-if (check_var('MACHINE', 'optiplex') || check_var('MACHINE', 'supermicro')) {
+if ((check_var('MACHINE', 'optiplex') || check_var('MACHINE', 'supermicro') || check_var('MACHINE', 'hpt630v1')) and (check_var('OS_INSTALL_LEGACY', '1'))) {
```
I think it should be moved below DRTM and BIOS kind checks, and then we could test just `OS_INSTALL_LEGACY` value, as all other platforms would be filtered out by BIOS/DRTM checks.

philipanda replied:
Like that?

https://github.com/3mdeb/openqa-tests-qubesos/compare/c1f63647bdb848a222fd9a6d97cbef1d5a4970c9..2ad98a32caefbfbb51ae8f4ff16a35a8ae849664#diff-5bcc1c5dc564aac714d0933cac3e9cb6d906077998db688ae82132d224af29acR71-R82

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -350,18 +452,31 @@ sub setup_aem {
     assert_script_run('rm -rf /var/lib/anti-evil-maid/aem');
 
-    assert_script_run("grub2-install $boot_disk");
+    # no need to install grub on EFI installations
```
Not quite, it still needs to be installed, but it is done automatically by RPM hooks for EFI, so maybe something like:
```suggestion
    # no need to manually install grub on EFI installations
```

krystian-hebel commented on generalhw/hpt630v1/power:
```diff
@@ -0,0 +1,106 @@
+# echo "$0: not doing anything"
+# exit 0
+
+function sonoff_pwr() {
```
We probably should add some delay after power off.

philipanda replied:
I haven't encountered any issues with powering the platform on/off, but it surely won't hurt.
It's important to not add any delay when powering on, or the screen with the HP logo might not get registered

https://github.com/3mdeb/openqa-tests-qubesos/compare/52d01ba%5E..52d01ba

krystian-hebel commented on generalhw/hpt630v1/README.md:
```diff
@@ -0,0 +1,50 @@
+
+These settings can be added to `openqa-cli` posting command to specify which
+packages to use.  Current default uses
+<https://dl.3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_v0.3/>.
```
Update according to whatever will be used in template. The same may be needed for other platforms, so maybe this should be made generic and point to template so we won't have to bump it with each new version.

philipanda replied:
https://github.com/3mdeb/openqa-tests-qubesos/compare/c4107f9%5E..c4107f9

I am not sure about MSI. The README and the instructions to start the tests are different than for other platforms.
I think it should probably still be fine to override the parameters like in the case of the other platforms.

krystian-hebel commented on generalhw/hpt630v1/ks.cfg:
```diff
@@ -0,0 +1,76 @@
+
+# allow all USB devices, this setting seems to appear on first boot, hence
+# the update is performed here
+sed -i -e 's/authorized_default=0/authorized_default=1/' /boot/grub2/grub.cfg /etc/default/grub
```
For this platform, we should also add `spec-ctrl=no-ibpb-entry` to Xen cmdline, otherwise VMs may not be able to start before timeout under some conditions. It will also make it much faster, and for testing we care more about time than about security implications this option has.

philipanda replied:
Does it look good?
https://github.com/3mdeb/openqa-tests-qubesos/compare/bba37a8%5E..bba37a8

krystian-hebel replied:
I think you need to escape `$` so it won't be parsed by shell executing `echo`. Maybe putting the whole string in single quotes and unescaping inner double quotes instead would be cleaner.

philipanda replied:
Right, it makes more sense if Grub interprets the variable.

https://github.com/3mdeb/openqa-tests-qubesos/compare/455291aad9cc99c2752ef8824ab92650daab4e20..d759221a46937501909ed0ad25f260e9530a568c

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -350,18 +452,31 @@ sub setup_aem {
     # cleanup in case AEM was previously initialized (useful for debugging this test)
```
(comment to code few lines above, but GH won't allow me to add it there)

Currently, `install_packages` fails on `dnf reinstall` step, because there is nothing to reinstall. Ideally we shouldn't use `dnf` directly, instead relying on `qubes-dom0-update`, as was done in blog posts. There still may be some install vs reinstall issues, and I don't know if `assert_script_run` is enough to catch all problems, like missing packages or version conflicts.

philipanda replied:
`assert_script_run` returns the command output, so there is no reason not to check it and use it to decide on fail/pass.
I'll see if using `qubes-dom0-update` helps and if there are some characteristic output meaning a success/fail.

philipanda replied:
@krystian-hebel 
In the blog post we first update dom0 before installing anything. The tests are not doing that. Should a general update to the whole dom0 be added?

krystian-hebel replied:
This is worth considering, but I don't think we should do it here and now. Let's focus on getting it to work in the first place.

philipanda replied:
Installing rpm files using qubes-dom0-update does not work out of the box. I've found that someone had a similar idea and it might be required for the "updatevm" to also have the files available on the same path as the qube on which the command is being called. https://forum.qubes-os.org/t/how-to-install-a-local-rpm-package-file-in-dom0/27737/3

There is no domain named "updatevm" though. Otherwise it might be possible to copy the rpm packages there like this https://forum.qubes-os.org/t/how-to-copy-files-between-vms-without-gui/8311/21

krystian-hebel replied:
You don't want to install rpm files directly, you should create a repository (according to our wiki), point to it in `/etc/yum.repos.d/aem.repo` and select that repo on calls to `qubes-dom0-update`, just as done in blog post. We could create signing key on the fly, or create it once and reuse it, just make sure it explicitly says that it isn't production key.

philipanda replied:
Adding the repository and using it with `qubes-dom0-update` added in https://github.com/3mdeb/openqa-tests-qubesos/compare/e6f3303%5E..e6f3303

Added a script to quickly setup a repo on the openqa host here https://github.com/3mdeb/openqa-tests-qubesos/compare/1ca9276%5E..1ca9276

krystian-hebel commented on generalhw/hpt630v1/power:
```diff
@@ -0,0 +1,106 @@
+    cmd=$(cut -d "p" -f2 <<< "$cmd" )
+    post_params="cmnd=Power $cmd"
+    echo "Sending sonoff POST request: $post_params\n"
+    curl -X POST http://192.168.10.157/cm -d "$post_params" && echo "\n"
```
Hardcoded IP, perhaps we should pass it as `$1` in place of `$rte_ip`, AFAICT we're not using it for anything.

philipanda replied:
Right. Now I see that, for example, supermicro also has a different meaning of the first parameter. It's a good idea then.

philipanda replied:
https://github.com/3mdeb/openqa-tests-qubesos/compare/75fb8a7%5E..75fb8a7

krystian-hebel reviewed (commented):
Do we need `v1` in the platform name? Ideally we should reuse the same setup for both platforms.

philipanda committed:
hpt630v1/power: Add delay after power off

to make sure the motherboard discharges completely

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
generalhw: Generalize hardcoded AEM packages version

It's defined in the templates. It is enough to point there

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda commented:
> Do we need v1 in the platform name? Ideally we should reuse the same setup for both platforms.

The two platforms (v1 and v2) have different versions of TPM chips. Other than that I think they are identical, and the tests don't need to take that into account. If it is not important to clearly differentiate them when scheduling / presenting the test results, then I think this suffix can be removed without problems.

philipanda committed:
hpt630v1/power: Use $1 for sonoff ip instead of hardcoding it

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
hpt630v1/ks.cfg: Add `spec-ctrl=no-ibpb-entry` for xen

Co-authored-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
aem_hw.pm: Fix comment on optional grub2-install call

Co-authored-by: Krystian Hebel <krystian.hebel@3mdeb.com>

philipanda committed:
aem_hw: Download AEM packages using qubes-dom0-update

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

aem_hw: Fix package names & repo definition

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
serve_aem_repo.sh: Add script to quickly setup testing AEM repo

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
3mdeb-templates: Modify aem_setup PACKAGES_BASE_URL

to work with the newest changes and installing using qubes-dom0-update

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
needles: grub after installing AEM

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -90,9 +95,15 @@ sub run {
 
         # using this to support re-installing AEM
-        assert_serial 'Welcome to GRUB!|Press enter to boot the selected OS';
+        assert_screen 'bootloader';
+        # assert_serial 'Welcome to GRUB!|Press enter to boot the selected OS';
         send_key 'end';
         send_key 'up';
+        # selects "advanced options"
+        send_key 'ret';
+        # selects xen hypervisor version
+        send_key 'ret';
+        # selects qubes version
         send_key 'ret';
```
This is different than before. Previously, we were selecting `Qubes, with Xen hypervisor` entry, not `Advanced options for ...`. This is because on UEFI there is additional entry for firmware settings. I haven't checked how it looks exactly, but it may use older version of Xen and/or kernel, which would explain different PCR values, and in turn error on unsealing.

```suggestion
        assert_screen 'bootloader';
        # assert_serial 'Welcome to GRUB!|Press enter to boot the selected OS';
        send_key 'end';
        # skip firmware settings on UEFI
        if (check_var('OS_INSTALL_LEGACY', '0'))
            send_key 'up';
        send_key 'up';
        send_key 'ret';
```

philipanda replied:
Oh, right, I haven't thought that there is an additional entry on the bottom back there.

This is bootentry is only selected during the setup of AEM though. In "firstboot" and "secondboot", when sealing and unsealing is checked, the default boot entry is selected. After that the default entry is being selected.

philipanda replied:
https://github.com/3mdeb/openqa-tests-qubesos/compare/436453b%5E..436453b

philipanda committed:
aem_hw: Fix aem_setup to not boot 'Advanced options'

Co-authored-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -342,26 +427,39 @@ sub install_packages {
     # doing it in one command.  Reinstallation happens first, because it just
```
Please update the comment after swapping the order.

philipanda replied:
Shouldn't I rather restore the order?
I think the comment still makes sense as qubes-dom0-update uses dnf internally.
https://github.com/3mdeb/openqa-tests-qubesos/compare/88ef02e8a354a6a8c480a1e88ec4a6267e45f413..751e0b758640250078983fbc94969a52fcf1cc05

krystian-hebel replied:
Does this order work? IIRC it returned failure because there were no packages to reinstall, but maybe I'm misremembering something.

philipanda replied:
It works fine, check: http://<openqa_IP>/tests/2197/logfile?filename=serial_terminal.txt
I've removed `anti-evil-maid`. It noticed that it can not reinstall it as it is not installed, but reinstalled all the other packages nevertheless. The `install` then only installed the missing `anti-evil-maid`

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -342,26 +427,39 @@ sub install_packages {
     # when this gets run initially.
-    assert_script_run("dnf reinstall -qy @to_install_reinstall");
-    assert_script_run("dnf install -qy @to_install_reinstall");
+    assert_script_run("qubes-dom0-update --disablerepo=\"*\" --enablerepo=aem --action=install -y @packages", timeout => 300);
+    assert_script_run("qubes-dom0-update --disablerepo=\"*\" --enablerepo=aem --action=reinstall -y @packages", timeout => 300);
```
Sidenote: this took ~7 minutes each for me, so the timeout should be increased to at least 10 minutes, if we were to remove `TIMEOUT_SCALE`.

philipanda replied:
Do you think it will take longer than 5 minutes on other devices too?
Why would we like to remove the `TIMEOUT_SCALE`?

HPt630 is so slow that the installation of QubesOS timed out and using this setting the timeout can be increased without affecting the tests on other devices.

Do we have an intention of ever merging the TB openqa fork into Qubes upstream? If so then I think using the `TIMEOUT_SCALE` would be safer. If not then why not just adjust all the timeouts to our needs?

krystian-hebel replied:
No, let's keep it, I just wanted to add it as a note marked as resolved (but couldn't) with no further action expected, at least for now.

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -265,51 +359,50 @@ sub setup_skl {
     assert_script_run "sed -i '/sinit_module_list=/s/\\*SINIT\\*\\.BIN/skl.bin/' /etc/grub.d/19_linux_xen_trenchboot";
```
This workaround is no longer needed and should be removed.

philipanda replied:
https://github.com/3mdeb/openqa-tests-qubesos/compare/88ef02e%5E..88ef02e

krystian-hebel replied:
Nope, that was another line that must stay.

philipanda replied:
Oops, there were three separate workarounds using `sed`, sorry!
Let's try again :sweat_smile: 
https://github.com/3mdeb/openqa-tests-qubesos/compare/751e0b758640250078983fbc94969a52fcf1cc05..ffdb8dbd323e7a5395cba23b11d3aeefbee7c96f

philipanda committed:
tests: shutdown.pm: Fix to work after aem_hw

aem_hw powered off the device which caused shutdown to fail
The device will now be shut down like in aem_hw in shutdown.pm if
aem_hw tests are being performed

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
needles/root-console-after-login-20250213: Add

The old ones matched only in ~96% on hpt630 which was on the
verge of working

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

krystian-hebel commented on generalhw/msi/README.md:
```diff
@@ -110,3 +110,13 @@ openqa-cli api -X POST isos ISO=Qubes-R4.2.0-rc1-x86_64.iso \
+
+* `PACKAGES_BASE_URL` - where to look for AEM-related packages.
+* `AEM_VER` - version of `anti-evil-maid` package
+* `SKL_VER` - version of "`ecure-kernel-loader` package
```
Typo, but also doesn't apply to MSI and can be removed (SKL is only for AMD).

philipanda replied:
Right
https://github.com/3mdeb/openqa-tests-qubesos/compare/5c6ee4cb7cad7d08f7fea6cb5fdc1aad6f45672d..88ef02e8a354a6a8c480a1e88ec4a6267e45f413

I've accidentally force-pushed the change in the history and a normal commit together. The compare show them both.

krystian-hebel reviewed (commented):
> clearly differentiate them when scheduling / presenting the test results

Fair point, let's leave it as is for now.

philipanda committed:
aem_hw: Remove unneeded aem skl workaround

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>
